plugins/source/aws/codegen/recipes/base.go

```diff
@@ -157,5 +185,77 @@ func (r *Resource) Generate() error {
 	if err := os.WriteFile(filePath, content, 0644); err != nil {
 		return fmt.Errorf("failed to write file %s: %w", filePath, err)
 	}
+
 	return nil
 }
+
+func (r *Resource) generateResolver(dir string) error {
+	tpl, err := template.New("fetch.go.tpl").Funcs(template.FuncMap{
+		"ToCamel": strcase.ToCamel,
+		"ToLower": strings.ToLower,
+	}).ParseFS(templatesFS,
+		fmt.Sprintf("templates/resolver_and_mock_test/%s/fetch.go.tpl", r.ResolverAndMockTestTemplate))
+	if err != nil {
+		return fmt.Errorf("failed to parse gcp templates: %w", err)
+	}
+
+	var buff bytes.Buffer
+	if err := tpl.Execute(&buff, r); err != nil {
+		return fmt.Errorf("failed to execute template: %w", err)
+	}
+
+	filePath := path.Join(dir, r.SubService+"_fetch.go")
+	content := buff.Bytes()
+	formattedContent, err := format.Source(buff.Bytes())
+	if err != nil {
+		fmt.Printf("failed to format source: %s: %v\n", filePath, err)
+	} else {
+		content = formattedContent
+	}
+
+	if err := os.WriteFile(filePath, content, 0644); err != nil {
+		return fmt.Errorf("failed to write file %s: %w", filePath, err)
+	}
+
+	return nil
+}
+
+func (r *Resource) generateMockTest(dir string) error {
+	tpl, err := template.New("mock_test.go.tpl").Funcs(template.FuncMap{
+		"ToCamel": strcase.ToCamel,
```

Comment:
Let's use our [own caser](https://github.com/cloudquery/plugin-sdk/blob/main/caser/caser.go#L144) here rather than github.com/iancoleman/strcase

Reply:
I simply copy-pasted from our schema-resolver :) Also, because the mock-test needs to reference the Table from the schema - important that they use the same function.

Changing this is out-of-scope, I think.